src/main/java/org/cryptomator/windows/quickaccess/ExplorerQuickAccessService.java

```diff
@@ -0,0 +1,150 @@
+package org.cryptomator.windows.quickaccess;
+
+import org.cryptomator.integrations.common.OperatingSystem;
+import org.cryptomator.integrations.common.Priority;
+import org.cryptomator.integrations.quickaccess.QuickAccessService;
+import org.cryptomator.integrations.quickaccess.QuickAccessServiceException;
+import org.cryptomator.windows.common.RegistryKey;
+import org.cryptomator.windows.common.WindowsException;
+import org.cryptomator.windows.common.WindowsRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.util.UUID;
+
+/**
+ * Implementation of the {@link QuickAccessService} for Windows Explorer
+ * <p>
+ * Based on a <a href="https://learn.microsoft.com/en-us/windows/win32/shell/integrate-cloud-storage">Microsoft docs example</a>.
+ */
+@Priority(100)
+@OperatingSystem(OperatingSystem.Value.WINDOWS)
+public class ExplorerQuickAccessService implements QuickAccessService {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ExplorerQuickAccessService.class);
+
+	@Override
+	public QuickAccessEntry add(Path target, String displayName) throws QuickAccessServiceException {
+		if (displayName == null) {
+			throw new IllegalArgumentException("Parameter 'displayname' must not be null.");
+		}
+		if (target == null) {
+			throw new IllegalArgumentException("Parameter 'target' must not be null.");
+		}
+		var entryName = "Vault - " + displayName;
+		var clsid = "{" + UUID.randomUUID() + "}";
+		LOG.debug("Creating navigation pane entry with CLSID {}", clsid);
+		//1. reg add HKCU\Software\Classes\CLSID\{0672A6D1-A6E0-40FE-AB16-F25BADC6D9E3} /ve /t REG_SZ /d "MyCloudStorageApp" /f
+		try (var t = WindowsRegistry.startTransaction()) {
+			try (var baseKey = t.createRegKey(RegistryKey.HKEY_CURRENT_USER, "Software\\Classes\\CLSID\\" + clsid, true)) {
+				baseKey.setStringValue("", entryName, false);
+
+				//2. reg add HKCU\Software\Classes\CLSID\{0672A6D1-A6E0-40FE-AB16-F25BADC6D9E3}\DefaultIcon /ve /t REG_EXPAND_SZ /d %%SystemRoot%%\system32\imageres.dll,-1043 /f
+				//TODO: should this be customizable?
+				try (var iconKey = t.createRegKey(baseKey, "DefaultIcon", true)) {
+					var exePath = ProcessHandle.current().info().command();
+					if(exePath.isPresent()) {
+						iconKey.setStringValue("", exePath.get(), false);
+					} else {
+						iconKey.setStringValue("", "%SystemRoot%\\system32\\shell32.dll,4", true); //the regular folder icon
```

Comment:
The Windows registry differentiates between different key types, two of them are `REG_SZ` and `REG_EXPAND_SZ`. The first one is a simple string, the second one is preprocessed before written to/loaded from the registry by replacing env variables.

Since in java there are only strings, i created only one method and added a parameter to indicate wether this string is expandable or not.